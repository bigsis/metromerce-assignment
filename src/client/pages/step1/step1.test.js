import React from 'react';
import { mount } from 'enzyme';
import Step1 from './step1';
import toJson from 'enzyme-to-json';

const mockInput = {
  0: [
    {
      id: 10,
      title: 'House',
      level: 0,
      children: [],
      parent_id: null
    }
  ],
  1: [
    {
      id: 12,
      title: 'Red Roof',
      level: 1,
      children: [],
      parent_id: 10
    },
    {
      id: 18,
      title: 'Blue Roof',
      level: 1,
      children: [],
      parent_id: 10
    },
    {
      id: 13,
      title: 'Wall',
      level: 1,
      children: [],
      parent_id: 10
    }
  ],
  2: [
    {
      id: 17,
      title: 'Blue Window',
      level: 2,
      children: [],
      parent_id: 12
    },
    {
      id: 16,
      title: 'Door',
      level: 2,
      children: [],
      parent_id: 13
    },
    {
      id: 15,
      title: 'Red Window',
      level: 2,
      children: [],
      parent_id: 12
    }
  ]
};

describe('Step1 page', () => {
  it('should be defined', () => {
    expect(Step1).toBeDefined();
  });

  it('should match snapshot when format json', () => {
    const wrapper = mount(<Step1 />, { attachTo: document.body });
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setState({ input: JSON.stringify(mockInput) });
    wrapper.find('Button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
