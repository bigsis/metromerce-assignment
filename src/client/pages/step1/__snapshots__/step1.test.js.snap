// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1 page should match snapshot when format json 1`] = `
<Step1>
  <div
    className="step1-content"
  >
    <div
      className="formatter"
    >
      <section
        className="input"
      >
        <Label
          for="exampleText"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <label
            className=""
            htmlFor="exampleText"
          >
            Input
          </label>
        </Label>
        <Input
          id="inputText"
          onChange={[Function]}
          rows="25"
          type="textarea"
          value=""
        >
          <textarea
            className="form-control"
            id="inputText"
            onChange={[Function]}
            rows="25"
            type="textarea"
            value=""
          />
        </Input>
      </section>
      <section
        className="output"
      >
        <Label
          for="exampleText"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <label
            className=""
            htmlFor="exampleText"
          >
            Output
          </label>
        </Label>
        <Input
          id="outputText"
          rows="25"
          type="textarea"
        >
          <textarea
            className="form-control"
            id="outputText"
            rows="25"
            type="textarea"
          />
        </Input>
      </section>
    </div>
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </Button>
  </div>
</Step1>
`;

exports[`Step1 page should match snapshot when format json 2`] = `
<Step1>
  <div
    className="step1-content"
  >
    <div
      className="formatter"
    >
      <section
        className="input"
      >
        <Label
          for="exampleText"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <label
            className=""
            htmlFor="exampleText"
          >
            Input
          </label>
        </Label>
        <Input
          id="inputText"
          onChange={[Function]}
          rows="25"
          type="textarea"
          value="{\\"0\\":[{\\"id\\":10,\\"title\\":\\"House\\",\\"level\\":0,\\"children\\":[],\\"parent_id\\":null}],\\"1\\":[{\\"id\\":12,\\"title\\":\\"Red Roof\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10},{\\"id\\":18,\\"title\\":\\"Blue Roof\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10},{\\"id\\":13,\\"title\\":\\"Wall\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10}],\\"2\\":[{\\"id\\":17,\\"title\\":\\"Blue Window\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":12},{\\"id\\":16,\\"title\\":\\"Door\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":13},{\\"id\\":15,\\"title\\":\\"Red Window\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":12}]}"
        >
          <textarea
            className="form-control"
            id="inputText"
            onChange={[Function]}
            rows="25"
            type="textarea"
            value="{\\"0\\":[{\\"id\\":10,\\"title\\":\\"House\\",\\"level\\":0,\\"children\\":[],\\"parent_id\\":null}],\\"1\\":[{\\"id\\":12,\\"title\\":\\"Red Roof\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10},{\\"id\\":18,\\"title\\":\\"Blue Roof\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10},{\\"id\\":13,\\"title\\":\\"Wall\\",\\"level\\":1,\\"children\\":[],\\"parent_id\\":10}],\\"2\\":[{\\"id\\":17,\\"title\\":\\"Blue Window\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":12},{\\"id\\":16,\\"title\\":\\"Door\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":13},{\\"id\\":15,\\"title\\":\\"Red Window\\",\\"level\\":2,\\"children\\":[],\\"parent_id\\":12}]}"
          />
        </Input>
      </section>
      <section
        className="output"
      >
        <Label
          for="exampleText"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <label
            className=""
            htmlFor="exampleText"
          >
            Output
          </label>
        </Label>
        <Input
          id="outputText"
          rows="25"
          type="textarea"
          value="[
    {
        \\"id\\": 10,
        \\"title\\": \\"House\\",
        \\"level\\": 0,
        \\"children\\": [
            {
                \\"id\\": 12,
                \\"title\\": \\"Red Roof\\",
                \\"level\\": 1,
                \\"children\\": [
                    {
                        \\"id\\": 17,
                        \\"title\\": \\"Blue Window\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 12
                    },
                    {
                        \\"id\\": 15,
                        \\"title\\": \\"Red Window\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 12
                    }
                ],
                \\"parent_id\\": 10
            },
            {
                \\"id\\": 18,
                \\"title\\": \\"Blue Roof\\",
                \\"level\\": 1,
                \\"children\\": [],
                \\"parent_id\\": 10
            },
            {
                \\"id\\": 13,
                \\"title\\": \\"Wall\\",
                \\"level\\": 1,
                \\"children\\": [
                    {
                        \\"id\\": 16,
                        \\"title\\": \\"Door\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 13
                    }
                ],
                \\"parent_id\\": 10
            }
        ],
        \\"parent_id\\": null
    }
]"
        >
          <textarea
            className="form-control"
            id="outputText"
            rows="25"
            type="textarea"
            value="[
    {
        \\"id\\": 10,
        \\"title\\": \\"House\\",
        \\"level\\": 0,
        \\"children\\": [
            {
                \\"id\\": 12,
                \\"title\\": \\"Red Roof\\",
                \\"level\\": 1,
                \\"children\\": [
                    {
                        \\"id\\": 17,
                        \\"title\\": \\"Blue Window\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 12
                    },
                    {
                        \\"id\\": 15,
                        \\"title\\": \\"Red Window\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 12
                    }
                ],
                \\"parent_id\\": 10
            },
            {
                \\"id\\": 18,
                \\"title\\": \\"Blue Roof\\",
                \\"level\\": 1,
                \\"children\\": [],
                \\"parent_id\\": 10
            },
            {
                \\"id\\": 13,
                \\"title\\": \\"Wall\\",
                \\"level\\": 1,
                \\"children\\": [
                    {
                        \\"id\\": 16,
                        \\"title\\": \\"Door\\",
                        \\"level\\": 2,
                        \\"children\\": [],
                        \\"parent_id\\": 13
                    }
                ],
                \\"parent_id\\": 10
            }
        ],
        \\"parent_id\\": null
    }
]"
          />
        </Input>
      </section>
    </div>
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </Button>
  </div>
</Step1>
`;
